libs/application/templates/car-recycling/src/lib/dataSchema.ts

```diff
@@ -1,16 +1,18 @@
 import { z } from 'zod'
 
-const Vehicles = z.object({
-  make: z.string(),
-  role: z.string(),
-  color: z.string(),
-  permno: z.string(),
-})
+const Vehicles = z
+  .object({
+    make: z.string(),
+    role: z.string(),
+    color: z.string(),
+    permno: z.string(),
+    mileage: z.string(),
+  })
+  .refine(({ mileage }) => mileage !== '' && mileage !== '0')
```

Comment:
Change dataSchema validation to be able to show the user what is causing the error
```suggestion
const Vehicles = z.object({
  make: z.string(),
  role: z.string(),
  color: z.string(),
  permno: z.string(),
  mileage: z.string().refine((m) => m !== '' && m !== '0'),
})
```

Reply:
fixed